include/internal/libspdm_common_lib.h

```diff
@@ -522,6 +524,62 @@ void libspdm_internal_dump_hex(const uint8_t *data, size_t size);
 #define LIBSPDM_INTERNAL_DUMP_DATA(data, size)
 #endif /* LIBSPDM_DEBUG_PRINT_ENABLE */
 
+/* Required scratch buffer size for libspdm internal usage.
+ * It may be used to hold the encrypted/decrypted message and/or last sent/received message.
+ * It may be used to hold the large request/response and intermediate send/receive buffer
+ * in case of chunking.
+ *
+ * If chunking is not supported, it should be at least below.
+ * +--------------------------+-----------------+-----------------+
+ * |    SENDER_RECEIVER       |MAX_SPDM_MSG_SIZE|MAX_SPDM_MSG_SIZE|
+ * +--------------------------+-----------------+-----------------+
+ * |<-Snd/Rcv buf for chunk ->|<-last request ->|<-cache request->|
+ *
+ *
+ * If chunking is supported, it should be at least below.
+ * +---------------+--------------+--------------------------+------------------------------+-----------------+-----------------+
+ * |SECURE_MESSAGE |LARGE_MESSAGE |    SENDER_RECEIVER       | LARGE SENDER_RECEIVER        |MAX_SPDM_MSG_SIZE|MAX_SPDM_MSG_SIZE|
+ * +---------------+--------------+--------------------------+------------------------------+-----------------+-----------------+
+ * |<-Secure msg ->|<-Large msg ->|<-Snd/Rcv buf for chunk ->|<-Snd/Rcv buf for large msg ->|<-last request ->|<-cache request->|
+ *
+ *
+ * The value is NOT configurable.
+ * The value MAY be changed in different libspdm version.
+ * It is exposed here, just in case the libspdm consumer wants to configure the setting at build time.
+ */
+#if LIBSPDM_ENABLE_CAPABILITY_CHUNK_CAP
+/* first section */
+uint32_t libspdm_get_scratch_buffer_secure_message_offset(libspdm_context_t *spdm_context);
```

Comment:
Easier and more scalable to just have `libspdm_get_scratch_buffer_offset` and `libspdm_get_scratch_buffer_capacity` functions that take an enum as input for what you want (`LIBSPDM_SCRATCH_SECURE_MESSAGE`, `LIBSPDM_SCRATCH_LARGE_MESSAGE`, etc)

Reply:
Technically, I can implement it in this way. See last 2 patches "adopt review feedback ..."

However, I think the original style is better, because it is simple and obvious.
Using switch/case to check enum is unnecessary to me.

I am not sure about what scalability issue we have with original style, since we can always add new internal functions for more section in scratch buffer.

Reply:
> I prefer we keep the original style.

That's fine.

Reply:
Then I drop the last 2 commit.